core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -150,6 +150,11 @@ private[spark] class DAGScheduler(
    */
   private[scheduler] val shuffleIdToMapStage = new HashMap[Int, ShuffleMapStage]
   private[scheduler] val jobIdToActiveJob = new HashMap[Int, ActiveJob]
+  /**
+   * Mapping from shuffle id to the indeterminate stage attempt id. Only includes the retried
+   * stages whose output deterministic level is indeterminate.
+   */
+  private[scheduler] val shuffleIdToIndeterminateStageAttemptId = new HashMap[Int, Int]
```

Comment:
why do we need this map? I think we just use the stage attempt id, right?

Reply:
We don't need this, it's old code remain, I'll delete this, sorry.

Reply:
Done in dc527dd.